nucypher-core-python/src/lib.rs

```diff
@@ -630,6 +631,126 @@ impl ReencryptionResponse {
     }
 }
 
+//
+// Threshold Decryption Request
+//
+
+#[pyclass(module = "nucypher_core")]
+#[derive(derive_more::From, derive_more::AsRef)]
+pub struct ThresholdDecryptionRequest {
+    backend: nucypher_core::ThresholdDecryptionRequest,
+}
+
+#[pymethods]
+impl ThresholdDecryptionRequest {
+    #[new]
+    pub fn new(
+        id: u16,
+        variant: u8,
+        ciphertext: &[u8], // TODO use ferveo Ciphertext type
+        conditions: Option<&Conditions>,
+        context: Option<&Context>,
+    ) -> Self {
+        let ferveo_variant: FerveoVariant;
+        match variant {
+            0 => ferveo_variant = FerveoVariant::SIMPLE,
+            1 => ferveo_variant = FerveoVariant::PRECOMPUTED,
+            _ => panic!("Invalid variant"),
+        }
+
+        Self {
+            backend: nucypher_core::ThresholdDecryptionRequest::new(
+                id,
+                ciphertext,
+                conditions
+                    .map(|conditions| conditions.backend.clone())
+                    .as_ref(),
+                context.map(|context| context.backend.clone()).as_ref(),
+                &ferveo_variant,
+            ),
+        }
+    }
+
+    #[getter]
+    pub fn id(&self) -> u16 {
+        self.backend.ritual_id
+    }
+
+    #[getter]
+    pub fn conditions(&self) -> Option<Conditions> {
+        self.backend
+            .conditions
+            .clone()
+            .map(|conditions| Conditions {
+                backend: conditions,
+            })
+    }
+
+    #[getter]
+    pub fn context(&self) -> Option<Context> {
+        self.backend
+            .context
+            .clone()
+            .map(|context| Context { backend: context })
+    }
+
+    #[getter]
+    pub fn ciphertext(&self) -> &[u8] {
+        self.backend.ciphertext.as_ref()
+    }
+
+    #[getter]
+    pub fn variant(&self) -> u8 {
+        match self.backend.variant {
+            FerveoVariant::SIMPLE => 0,
+            FerveoVariant::PRECOMPUTED => 1,
+        }
+    }
```

Comment:
What do you think about exposing the actual enum instead of raw bytes here?

Reply:
I like that idea, is it possible to expose enums directly?

Reply:
No built-in support yet, but there are some examples here: https://github.com/PyO3/pyo3/issues/417#issuecomment-1431099651